caas/kubernetes/provider/admissionregistration_test.go

```diff
@@ -42,8 +43,11 @@ func (s *K8sBrokerSuite) assertMutatingWebhookConfigurations(c *gc.C, cfgs []k8s
 	numUnits := int32(2)
 	statefulSetArg := &appsv1.StatefulSet{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:   "app-name",
-			Labels: map[string]string{"juju-app": "app-name"},
+			Name: "app-name",
+			Labels: utils.LabelsMerge(
+				utils.LabelsForApp("app-name", false),
+				utils.LabelsJuju,
+			),
```

Comment:
I don't agree with the assertion. Hard coding expectation is what dragged this PR into a very long time as lots of tests had to be updated. It would be far better and easier to write a single set of tests that check the output from these methods matches a given set of expectations.

Because we use mock's it means we end up testing a lot of what we don't care about i.e above. It would be far better to do the work to assert that what we care about is there and not verbatim check the whole object.

Reply:
@wallyworld I agree with @tlm here, we shouldn't be retesting logic. Once a set of functionality has sufficient unit tests, its uses elsewhere are considered tested. By hard-coding the labels values in every unrelated test, we just shoot ourselves in the foot.